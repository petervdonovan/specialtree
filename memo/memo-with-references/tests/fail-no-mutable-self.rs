use memo_with_references::memo;

struct Foo;

impl Foo {
    #[memo('a)]
    fn memome<'a>(&mut self) {}
}

fn main() {}
