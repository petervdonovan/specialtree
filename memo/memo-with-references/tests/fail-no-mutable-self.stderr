error: memoizable functions cannot have mutable self parameter
 --> tests/fail-no-mutable-self.rs:7:19
  |
7 |     fn memome<'a>(&mut self) {}
  |                   ^^^^^^^^^
